client/client.go

```diff
@@ -244,6 +244,15 @@ func WithForwardingOption(enableForwarding bool) ClientOption {
 	}
 }
 
+// WithTSOServerProxyOption configures the client to use TSO server proxy,
+// i.e., the client will send TSO requests to the API leader (the TSO server
+// proxy) which will forward the requests to the TSO servers.
+func WithTSOServerProxyOption(useTSOServerProxy bool) ClientOption {
```

Comment:
Is it only used for testing?

Reply:
nope. If tso service discovery has problem, the new client can use this option to bypass tso service discovery too.

Reply:
the implementation has production quality.

Reply:
To let tool to support "tso server proxy", let me know if you have better idea.

Reply:
> nope. If tso service discovery has problem, the new client can use this option to bypass tso service discovery too.

If so, how do we set it? BTW, I only find it is used in the tso bench.

Reply:
provide a potential solution for any caller to use it. Just like we have provided followerProxy option

Reply:
it's mostly used by tool. Are you saying this reason isn't strong enough to add this option and if we add this option it could be abused by the client users?

Reply:
Do you have any better idea to let pd-tso-bench to send tso requests to  api leader instead of tso service directly? Or are you saying that we shouldn't let new PD client to talk to API leader to get TS? In my opinion, this is something that we let the new PD client to provide a legacy client to to talk to PD/API leader to get TS.

Reply:
IMO, if we provide such an option, anyone can use it. On the other hand, other components that use the pd client aren't going to support it for now. if we are going to test forward, we can call the API that the real scene will call, or use an old version of pd client.

Reply:
so I haven't get your point. What's your concern? Are you saying that if pd-tso-bench is the only one who needs this option then we shouldn't add it to pollute PD interface?

Reply:
Yes

Reply:
make sense. We can sync further on this and see if adding comment is sufficient or if there is better way.

Reply:
remove use tso server proxy option from this pr for now